dcpy/utils/versions.py

```diff
@@ -255,6 +255,56 @@ def is_newer(version_1: str, version_2: str) -> bool:
     return version_1_obj > version_2_obj
 
 
+def group_versions_by_base(version: str, versions_list: list[str]) -> list[str]:
+    """
+    Groups versions by their base, ignoring patch differences, and returns a list of matching versions
+    in ascending order (from oldest to newest).
+
+    Example:
+
+    ```python
+    version = '24Q1'
+    versions_list = ['24Q1', '24Q1.1', '24v3', '24v3.0.1', '23Q1']
+
+    matching_versions = group_versions_by_base(version, versions_list)
+    print(matching_versions)  # Output: ['24Q1', '24Q1.1']
+    ```
+    """
+    parsed_version = parse(version)
+    version_list_wo_latest = [v for v in versions_list if v != "latest"]
+    same_version_type_lst = [
+        v for v in version_list_wo_latest if isinstance(parse(v), type(parsed_version))
+    ]
+
+    def is_matching_version(base_version: Version, compare_version: Version) -> bool:
```

Comment:
Should we maybe just have `Version` have `patch` explicitly? Then we could do something a little more naive, like set version to zero for each and then just use `==`.

That being said, maybe it sort of complicates other things (need to ensure that dataclasses with inherited attributes still sort/compare properly). Seems like more a headache to change. So more just food for thought. Let's leave this as is!

Reply:
Or maybe version could have a property called `base_fields`, which returns the dict put out by the helper within this helper? Just spitballing

Reply:
I feel like maybe that's better taken care of when we overhaul versions a bit. Which we should probably do sometime soon 😬